mailbox/noise_patterns.go

```diff
@@ -0,0 +1,101 @@
+package mailbox
+
+var (
+	// XXPattern represents the XX Noise pattern using SPAKE2 to mask
+	// the ephemeral key of the first message.
+	// -> me
+	// <- e, ee, s, es
+	// -> s, se
+	XXPattern = HandshakePattern{
+		Name: "XXeke+SPAKE2",
+		Pattern: []MessagePattern{
+			{
+				Tokens:    []Token{me},
+				Initiator: true,
+				ActNum:    act1,
+			},
+			{
+				Tokens:    []Token{e, ee, s, es},
+				Initiator: false,
+				ActNum:    act2,
+			},
+			{
+				Tokens:    []Token{s, se},
+				Initiator: true,
+				ActNum:    act3,
+			},
+		},
+	}
+
+	// KKPattern represents the KK Noise pattern in which both initiator
```

Comment:
Noice, also this works as both sides already know which mailbox ID to be expecting, so they know which keys to expect.